core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/preferences/TitleBlockPreferencePage.java

```diff
@@ -0,0 +1,496 @@
+    top_layout.numColumns = 2;
+    top_layout.marginHeight = 2;
+    top_layout.marginWidth = 2;
+    top.setLayout(top_layout);
+    top.setLayoutData(new GridData(GridData.FILL_BOTH));
+
+    createTable(top);
+
+  }
+
+  private void createCheckBox() {
+    defaultTitleBlockFieldEditor = new PreferenceField("defaultTitleBlock", "Add by default Diagram TitleBlock",
+        getFieldEditorParent());
+    addField(defaultTitleBlockFieldEditor, UserProfileModeEnum.Expert, getFieldEditorParent(), ProjectScope.class);
+  }
+
+  private void createTable(Composite top) {
+    v = new TableViewer(top, SWT.MULTI | SWT.BORDER | SWT.H_SCROLL | SWT.V_SCROLL | SWT.FULL_SELECTION);
+
+    createColumns(v, columnsNumber);
+    v.setContentProvider(ArrayContentProvider.getInstance());
+    v.setInput(createModel(columnsNumber, rowsNumber));
+
+    v.getTable().setLinesVisible(true);
+    // v.getTable().setBackground(getShell().getDisplay().getSystemColor(SWT.COLOR_GREEN));
+    v.getTable().setHeaderVisible(true);
+    GridData gridData = new GridData(GridData.FILL_BOTH);
+    gridData.grabExcessVerticalSpace = true;
+    gridData.grabExcessHorizontalSpace = true;
+    gridData.verticalAlignment = GridData.FILL;
+    gridData.horizontalAlignment = GridData.FILL;
+    gridData.widthHint = 200;
+    gridData.horizontalSpan = 1;
+    v.getTable().setLayoutData(gridData);
+
+    addMenu(v);
+
+    Listener treeListener = new Listener() {
+      @Override
+      public void handleEvent(Event event) {
+        switch (event.type) {
+        case SWT.MouseDoubleClick: {
+          Point coords = new Point(event.x, event.y);
+
+          ViewerCell cell = v.getCell(coords);
+          int index = cell.getColumnIndex();
+          List<TitleBlockCell> list = (List<TitleBlockCell>) cell.getElement();
+          String name = list.get(index).name;
+          String content = list.get(index).content;
+          TitleBlockDialog dialog = new TitleBlockDialog(getShell());
+          if (content == "content") {
+            dialog.setCurrentName("");
+            dialog.setCurrentContent("");
+          } else {
+            dialog.setCurrentName(name);
+            dialog.setCurrentContent(content);
+          }
+          dialog.create();
+          if (dialog.open() == Window.OK) {
+            list.get(index).name = dialog.getName();
+            list.get(index).content = dialog.getContent();
+            cell.setText(dialog.getName());
+
+          }
+
+        }
+        }
+
+      }
+    };
+
+    v.getTable().addListener(SWT.MouseDoubleClick, treeListener);
+  }
+
+  private void disposeColumns() {
+    TableColumn[] columns = v.getTable().getColumns();
+    for (TableColumn tc : columns) {
+      tc.dispose();
+    }
+  }
+
+  private void refreshTableColumns() {
+    disposeColumns();
+    createColumns(v, columnsNumber);
+    v.setInput(updateModel(columnsNumber, rowsNumber));
+
+  }
+
+  private void refreshTableRows() {
+    rowsNumber += 1;

```

Comment:
why rowsNumber is increased? maybe rename the method from refreshTableRows to addNewTableRow.

Reply:
Done